app/multilistener.go

```diff
@@ -0,0 +1,74 @@
+package app
+
+import (
+	"context"
+	"net"
+
+	"github.com/pkg/errors"
+	"github.com/target/goalert/util/log"
+)
+
+type multiListener struct {
+	listeners []net.Listener
+
+	ch    chan net.Conn
+	errCh chan error
+}
+
+func newMultiListener(ln ...net.Listener) *multiListener {
+	ml := multiListener{listeners: ln}
+	ml.ch = make(chan net.Conn)
+	ml.errCh = make(chan error)
+
+	for _, l := range ln {
+		go ml.listen(l)
```

Comment:
Maybe add a `sync.WaitGroup` so that when `Close` is called, we wait for the background goroutines to finish. That way we can add a unit test that will validate proper cleanup too.